src/gui/dialogs/addon/manager.cpp

```diff
@@ -677,7 +726,44 @@ boost::dynamic_bitset<> addon_manager::get_type_filter_visibility() const
 				);
 			res.push_back(toggle_states[index]);
 		}
+		return res;
+	}
+}
+
+boost::dynamic_bitset<> addon_manager::get_lang_filter_visibility() const
+{
+	const multimenu_button& lang_filter = find_widget<const multimenu_button>(get_window(), "language_filter", false);
+
+	boost::dynamic_bitset<> toggle_states = lang_filter.get_toggle_states();
+	if(toggle_states.none() || toggle_states[0] == true) {
+		// Nothing or English (GB) selected. It means that *all* add-ons are shown.
+		boost::dynamic_bitset<> res_flipped(addons_.size());
+		return ~res_flipped;
+	} else {
+		boost::dynamic_bitset<> res;
 
+		for(const auto& a : addons_) {
+			bool retval = false;
+			// If English is selected and no translations are indicated in the addon, it is shown
+			if (toggle_states[0] == true && a.second.locales.empty()) {
+				retval = true;
+			}
+			// langcode -> string conversion vector, to be able to detect either
+			// langcodes or langstring entries
+			std::vector<std::string> lang_string_vector;
+			for (long unsigned int i = 0; i < a.second.locales.size(); i++) {
+				lang_string_vector.push_back(langcode_to_string(a.second.locales[i]));
+			}
+			// Find all toggle states, where toggle = true and lang = lang
+			for (long unsigned int i = 0; i < toggle_states.size(); i++) {
+				if (toggle_states[i] == true) {
+					if ((std::find(a.second.locales.begin(), a.second.locales.end(), language_filter_types_[i].second) != a.second.locales.end() ? true : false) || (std::find(lang_string_vector.begin(), lang_string_vector.end(), language_filter_types_[i].second) != lang_string_vector.end() ? true : false)) {
```

Comment:
Format this on more than one line.

Reply:
Looks like it could be simplified with `utils::contains`, which just needs src/utils/general.hpp to be `#included`.

Reply:
utils::contains was a great tip, reduced reading complexity of the code greatly :ok_hand: